### PR TITLE
Move `PathFilters` from detekt-api to detekt-utils

### PR DESCRIPTION
**Pull request by BraisGabin**
Also see #6814

~We had a little mess around this class. I splitted some tests and move them to the correct locations and then move `PathFilters` to `:detekt-tooling`.~

~We have far too much "utils" subprojects and I'm not 100% sure the responsability of which one, but this one seems the correct one. We don't want this at `:detekt-api` because it is only used by `:detekt-core` and by `:detekt-cli`. And it would be really nice to move it to `:detekt-core` but, as I said it is also used by `:detekt-cli`.~

~If someone create some documentation about which is the responsabilities of each subproject it would be great.~

Move `PathFilters` to `:detekt-utils`. This class is used on `detekt-cli` and `detekt-core` but we don't want it to be part of our public API so I'm moving it to `:detekt-utils`.

~Waiting for #7009~

### CODE REVIEW

File: detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/CliArgs.kt [hunk 1]
```diff
@@ -214,12 +214,4 @@ class CliArgs {
                 FailureSeverity.Info -> FailOnSeverity(minSeverity.toSeverity())
             }
         }
-
-    companion object {
```
Comment (3flex) [inline detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/CliArgs.kt:218]: I agree to remove this, but it should be in a different PR, this is unrelated to moving PathFilters around

**Comment by 3flex** [comment]
Thinking about this, I don't think the includes/excludes belongs in detekt-tooling at all, and it makes sense to remove them from `ProjectSpec`. Calls to start an analysis session should include the list of files to scan, but the filtering of those should be done by whatever is calling tooling. Different clients will have different methods of filtering file lists.

As an example, the Gradle plugin doesn't pass exclude rules. It already filters and just sends the file list to the CLI. I think it makes more sense to move the path filters to CLI, and remove includes/excludes from tooling and bring everything up a level.

What do you think?

File: detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/CliArgs.kt (hunk 1 above)
Comment (BraisGabin) [reply detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/CliArgs.kt]: I moved this to #6997. Right now it doesn't compiles because it needs the changes on this PR, I'll rebase it once this is merged.

**Comment by BraisGabin** [comment]
> What do you think?

That you are completely right. I'm moving this PR to draft to block it an I'll create another PR removing includes/excludes from `ProjectSpec`.

**Comment by BraisGabin** [comment]
I rebased this on top of #7009.

Now instead of moving `PathFilters` to `detekt-tooling` I move it to `detekt-utils`. We still use `PathFilters` on `detetk-cli` and `detekt-core` but now it is not part of any API, it is just an implementation detail.

I'm not a big fan of `detekt-utils` but it is handy to share code between parts of detekt without making it public API.

**Comment by BraisGabin** [comment]
This PR changed a bit since the last review. @3flex can you take a look again?

**Comment by 3flex** [comment]
> I'm not a big fan of detekt-utils but it is handy to share code between parts of detekt without making it public API.

Completely agree - it's useful though unless someone comes up with a better idea. LGTM after merge conflict is addressed.

**Review by 3flex — APPROVED** [review]
